maskwatch-bot/__init__.py

```diff
@@ -95,6 +96,19 @@ async def _is_oper(self, nickname: str):
         })
         return whois_line.command == RPL_WHOISOPERATOR
 
+    async def _format(self, string: str):
+        # expand reason templates
+        # this bit might be a little janky
```

Comment:
only a little, but it's probably how I'd have done it too 🐈

Reply:
after a bit of thought, what I might instead do is
```python3
for key, value in self._reasons.items():
    string = string.replace(f"${key}", value)
```

not sure if that's more likely to wrongly replace stuff than your solution though.

Reply:
ah actually that's not recursive is it

Reply:
yeah that's the reason for the while loop. might be worth changing the split() stuff though. that way for sure looks nicer.

Reply:
With the while loop bit, it works the same as before except a lot more efficient. I think we'll go with that.